### PR TITLE
unexpected result when use `in` in hive query

### PR DESCRIPTION
**Pull request by lgbo-ustc**
Changelog category (leave one):

- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Unexpected result when use `in` in `where` in hive query.

with following table
```
CREATE TABLE default.demo_parquet_parts (`id` Nullable(String), `score` Nullable(Int32), `day` Nullable(String), `hour` String) ENGINE = Hive('thrift://hivetest:9083', 'test', 'parquet_demo') PARTITION BY(day, hour);
```

the query's result is zero which is expected 2
```
SELECT count(*) FROM default.demo_parquet_parts where day = '2021-11-05' and hour in ('00')
```

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: docker/test/integration/hive_server/Dockerfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ COPY hadoop-env.sh /hadoop-3.1.0/etc/hadoop/
 #COPY core-site.xml /hadoop-3.1.0/etc/hadoop
 COPY core-site.xml.template /hadoop-3.1.0/etc/hadoop
 COPY hive-site.xml /apache-hive-2.3.9-bin/conf
+RUN echo "123"
```
Comment (kssenii) [inline docker/test/integration/hive_server/Dockerfile:41]: remove?

**Review by kssenii — APPROVED** [review]

File: docker/test/integration/hive_server/Dockerfile (hunk 1 above)
Comment (lgbo-ustc) [reply docker/test/integration/hive_server/Dockerfile]: Ok

**Comment by kssenii** [comment]
@lgbo-ustc integration test `test_hive_query` failed.

**Comment by lgbo-ustc** [comment]
> @lgbo-ustc integration test `test_hive_query` failed.

strange, running on local is fine. It looks like the hive server failed.
I will check it.

**Comment by lgbo-ustc** [comment]
> 

Hi, @kssenii, Does the docker image for test rebuild auto automatically？

**Comment by Felixoid** [comment]
It's updated, but not used in the tests by some reason https://s3.amazonaws.com/clickhouse-test-reports/34945/21c164ad2c63d8d73f7bc78e173d7a26034524fe/integration_tests__thread__actions__[1/4]/main_script_log.txt:

```
Kill one or more running containers
Unable to find image 'clickhouse/integration-tests-runner:latest' locally
latest: Pulling from clickhouse/integration-tests-runner
```

**Comment by lgbo-ustc** [comment]
> It's updated, but not used in the tests by some reason [https://s3.amazonaws.com/clickhouse-test-reports/34945/21c164ad2c63d8d73f7bc78e173d7a26034524fe/integration_tests__thread__actions__[1/4]/main_script_log.txt](https://s3.amazonaws.com/clickhouse-test-reports/34945/21c164ad2c63d8d73f7bc78e173d7a26034524fe/integration_tests__thread__actions__%5B1/4%5D/main_script_log.txt):
> 
> ```
> Kill one or more running containers
> Unable to find image 'clickhouse/integration-tests-runner:latest' locally
> latest: Pulling from clickhouse/integration-tests-runner
> ```

Yes. part of the test cases passed, but the ones which use the new hive table failed. The new hive table only exists in the new hive server image.  So I think the integration test image in used is not the latest one.

**Pull request merged**